hadoop-ozone/recon/src/main/java/org/apache/hadoop/ozone/recon/solr/SolrUtil.java

```diff
@@ -0,0 +1,492 @@
+  }
+
+  private void validateStartDate(String startDate,
+                                 List<NameValuePair> urlParameters) {
+    if (!StringUtils.isEmpty(startDate)) {
+      ZonedDateTime lastXUnitsOfZonedDateTime = null;
+      startDate = validateStartDate(startDate);
+      if (null != LastXUnit.getType(startDate)) {
+        lastXUnitsOfZonedDateTime =
+            lastXUnitsOfTime(LastXUnit.getType(startDate));
+      } else {
+        lastXUnitsOfZonedDateTime =
+            epochMilliSToZDT(startDate);
+      }
+      urlParameters.add(new BasicNameValuePair("fq",
+          setDateRange("evtTime",
+              Date.from(lastXUnitsOfZonedDateTime.toInstant()), null)));
+    }
+  }
+
+  private void validateAndAddSolrReqParam(
+      String paramVal, String paramName,
+      List<NameValuePair> urlParameters) {
+    if (!StringUtils.isEmpty(paramVal)) {
+      StringBuilder sb = new StringBuilder(paramName);
+      sb.append(":");
+      sb.append(paramVal);
+      urlParameters.add(new BasicNameValuePair("fq", sb.toString()));
+    }
+  }
+
+  private HttpRequestWrapper prepareHttpRequest(
+      List<NameValuePair> urlParameters, InetSocketAddress solrAddress,
+      String uri) {
+    // add request parameter, form parameters
+    urlParameters.add(new BasicNameValuePair("q", "*:*"));
+    urlParameters.add(new BasicNameValuePair("wt", "json"));
+    urlParameters.add(new BasicNameValuePair("fl",
+        "access, agent, repo, resource, resType, event_count"));
+    urlParameters.add(new BasicNameValuePair("fq", "access:read"));
+    urlParameters.add(new BasicNameValuePair("fq", "repo:cm_ozone"));
+
+    urlParameters.add(new BasicNameValuePair("sort", "event_count desc"));
+    urlParameters.add(new BasicNameValuePair("start", "0"));
+    urlParameters.add(new BasicNameValuePair("rows", "0"));
+
+    urlParameters.add(new BasicNameValuePair("json.facet", "{\n" +
+        "    resources:{\n" +
+        "      type : terms,\n" +
+        "      field : resource,\n" +
+        "      sort : \"read_access_count desc\",\n" +
+        "      limit : 100,\n" +
+        "      facet:{\n" +
+        "        read_access_count : \"sum(event_count)\"\n" +
+        "      }\n" +
+        "    }\n" +
+        "  }"));
+    HttpRequestWrapper requestWrapper =
+        new HttpRequestWrapper(solrAddress.getHostName(),
+            solrAddress.getPort(), uri,
+            urlParameters, HttpRequestWrapper.HttpReqType.POST);
+    return requestWrapper;
+  }
+
+  private void generateHeatMap(EntityMetaData[] entities) {
+    EntityReadAccessHeatMapResponse rootEntity =
+        entityReadAccessHeatMapRespRef.get();
+    rootEntity.setMinAccessCount(entities[0].getReadAccessCount());
+    rootEntity.setLabel("root");
+    List<EntityReadAccessHeatMapResponse> children =
+        rootEntity.getChildren();
+    Arrays.stream(entities).forEach(entityMetaData -> {
+      String path = entityMetaData.getVal();
+      String[] split = path.split("/");
```

Comment:
split size not validated for adding to bucket, if no "/" can cause OutOfIndex exception

Reply:
yes  added check to skip that iteration.